Sources/HaCWebsiteLib/Workshop.swift

```diff
@@ -0,0 +1,103 @@
+import Foundation
+import Yaml
+import KituraMarkdown
+
+public struct Workshop {
+  let title: String
+  let authors: [String]
+  let presenters: [String]
+  let thanks: [Thanks]
+  let recommendations: [Recommendation]
+  let links: [URL]
+  let tags: [String]
+  let prerequisites: String
+  let description: String
+
+  struct Thanks: Equatable {
+    let to: String
+    let reason: String?
+  }
+
+  struct Recommendation: Equatable {
+    let title: String
+    let url: URL
+  }
+
+  enum WorkshopError : Error {
+    case malformedMetadata
+  }
+}
+
+// Satisfy equatability for Workshop.Recommendation
```

Comment:
It feels like we're locking ourselves into writing a comment explaining this every time we need to conform to that protocol. This opens us up to people forgetting, convention on the comments, multiple sources of truth etc...

This seems like pretty idiomatic Swift to me, and as a non-Swiftie, was fairly clear.

However, if clarity is a concern, is it clearer to use the example that Apple gives on the docs? https://developer.apple.com/reference/swift/equatable shows that we can add the comparison as a static function on the struct:

> To adopt the Equatable protocol, implement the equal-to operator (==) as a static method of your type.

This will bring the equality function much closer to the struct and make the intent clearer.